src/UpgradeExecRouterBuilder.sol

```diff
@@ -142,38 +159,43 @@ contract UpgradeExecRouterBuilder {
             }
         }
 
-        // schedule that
+        // batch those calls to execute from the l1 timelock
         bytes memory timelockCallData = abi.encodeWithSelector(
             L1ArbitrumTimelock.scheduleBatch.selector,
             schedTargets,
             schedValues,
             schedData,
-            DEFAULT_PREDECESSOR, // CHRIS: TODO: should we allow others here? could be important? no, use the complex route?
+            predecessor,
             timelockSalt,
             l1TimelockMinDelay
         );
 
+        // withdraw to the l1 timelock
         return (
             address(100),
             abi.encodeWithSelector(ArbSys.sendTxToL1.selector, l1TimelockAddr, timelockCallData)
         );
     }
-    /// @notice creates data for ArbSys for a batch of core governance operations, using common default values for calldata and callvalue
-    /// @param chainIds target chain ids for actions
-    /// @param actionAddresses address of action contracts (on their target chain)
-    /// @param timelockSalt salt for core gov l1 timelock operation
 
+    /// @notice Creates the to address and calldata to be called to execute a route to an action contract
+    ///         Uses common defaults using for value, calldata and predecessor
+    ///         See Governance Action Contracts for more details
+    /// @param chainIds         Chain ids containing the actions to be called
+    /// @param actionAddresses  Addresses of the action contracts to be called
+    /// @param timelockSalt     A salt for the l1 timelock operation
     function createActionRouteDataWithDefaults(
         uint256[] memory chainIds,
         address[] memory actionAddresses,
-        bytes32 timelockSalt // CHRIS: TODO: can we calculate this in the contract somehow?
+        bytes32 timelockSalt // CHRIS: TODO: should this be a string instead of bytes32? basically this comes down to which is more painful - posting a big description into the tally ui, or posting it seomwhere else, hashing it and then pasting the result into tally?
```

Comment:
the description string would need to be pasted regardless, so assuming client-side support the UX should essentially be the same? should prob ask tally